test/cql-pytest/test_compaction_tombstone_gc.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from util import new_materialized_view, new_test_table
+import nodetool
+import time
+
+def test_tombstone_gc_with_conflict_in_memtable(scylla_only, cql, test_keyspace):
+    """
+    Regression test for fixed https://github.com/scylladb/scylladb/issues/20423
+    """
+    schema = "k int, v int, primary key (k, v)"
+    with new_test_table(cql, test_keyspace, schema, extra="with gc_grace_seconds = 0") as table:
+        with nodetool.no_autocompaction_context(cql, table):
+            nodetool.setlogginglevel(cql, "compaction", "trace")
+            # Insert initial data into the base table:
+            # Row 1 is expected to be garbage collected after expiration
+            cql.execute(f"insert into {table} (k, v) values (1, 1) using timestamp 1 and ttl 1")
```

Comment:
Please correct me if I'm wrong, but by doing flush() at the right time you could have written a simpler test without TTL, and avoid the problems of sleeps.
But if testing TTL is part of the *goal* of this test, please mention it (and explain why) in the comment introducing the test.

Reply:
> Please correct me if I'm wrong, but by doing flush() at the right time you could have written a simpler test without TTL, 

maybe, yes

> and avoid the problems of sleeps.

It won't avoid them entirely.
In the unit test we move the clock forward using `forward_jump_clocks` so it's not a problem.
In cql-pytest we can't do that so I used time.sleep(), but I'm not convinced it's a problem either.

> But if testing TTL is part of the _goal_ of this test, please mention it (and explain why) in the comment introducing the test.

It is not tested elsewhere so I do want to test it here.
I can add a comment about it.

Reply:
The sleep() is not incorrect, but it takes time. Maybe you're used to the world of dtest, or even topology tests, where a one-minute test is normal. I'm used to the world of test/alternator, where a thousand tests take half a minute. cql-pytest is in the middle - the whole suite takes a bit over a minute and not half a minute, but every full second of sleep hurts my eyes. For TTL tests, it's unfortunately unavoidable which is why I'm asking why you wanted this to be a TTL test, when compaction isn't about TTL. For the remaining 1 second of sleep about gc_grace_seconds, that's very surprising for me, more on this below.

Reply:
@raphaelsc?

Reply:
I also don't understand why we want a ttl test here?